File: src/main/java/io/gaegul/ch01/MallardDuck.java
package io.gaegul.ch01;

import io.gaegul.ch01.behavior.Flyable;
import io.gaegul.ch01.behavior.Quackable;

public class MallardDuck extends Duck implements Flyable, Quackable {

	@Override
	public void display() {
		System.out.println("청둥오리");
	}

	@Override
	public void fly() {
		System.out.println("오리날다~!");
	}

	@Override
	public void quack() {
		System.out.println("꽥꽥~");
	}
}

File: src/main/kotlin/io/gaegul/ch01/MallardDuck.kt
package io.gaegul.ch01

import io.gaegul.ch01.behavior.FlyWithWings
import io.gaegul.ch01.behavior.Quack

class MallardDuck: Duck(
    Quack(), FlyWithWings()
) {

    override fun display() {
        println("청둥오리")
    }
}